From f98301605a517478ec17287cd9c72574c16b56c4 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Fri, 2 Oct 2020 16:09:58 +0000
Subject: [PATCH 0951/1000] MANTIS 0058817 Installation: LGW Hybrid: 2 barcode
 scanners platform.idl V5.1

add mEnvironmentComponent
  realComponentNameBase
  unitId
  unitClass
add ComponentIF
  checkBindingConflict
add PlatformIF
  getBindingConflict
  registerDevice
  deregisterDevice

allow to resolve name conflict while binding components


git-svn-id: svn://localhost/SelfServiceCommon/trunk@1593 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Massai/platform.idl            | 63 +++++++++++--------
 1 file changed, 36 insertions(+), 27 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/platform.idl b/SelfServiceCommon/Massai/idl/Massai/platform.idl
index 346692ee..a7f0bd49 100644
--- a/SelfServiceCommon/Massai/idl/Massai/platform.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/platform.idl
@@ -27,7 +27,33 @@
 
 module PlatformManagement
 {
-    const string interface_version = "5.0";
+    const string interface_version = "5.1";
+
+    /** Platform interface for the device components. Via this interface the device
+        components communicate with the platform manager internally. */
+    enum LoggingClass
+    {
+      LC_APPLICATION,
+      LC_PRINTER,
+      LC_SCANNER,
+      LC_CONVEYOR,
+      LC_BIN1,
+      LC_BIN2,
+      LC_BIN3,
+      LC_BIN4,
+      LC_CAPTURE_PRINTER,
+      LC_CAPTURE_READER,
+      LC_DISPENSER_PRINTER,
+      LC_DISPENSER_READER,
+      LC_PRINTER_FEEDER,
+      LC_CONVEYOR_FEEDER,
+      LC_DISK,
+      LC_NETWORK,
+      LC_UNDEFINED,
+      LC_PLF,
+      LC_INDICATOR,
+      LC_AUDIO
+    };
 
     /** Data type for text lists. */
     typedef sequence<string> list;
@@ -68,6 +94,9 @@ module PlatformManagement
                                           at which position in the component list
                                           the linked components can be found.
                                           (Index counting starts with 0) */
+        string realComponentNameBase; /**< Real component name(e.g. ATB_PRNT_BP, ATB_PRNT_TK) */
+        string unitId;                /**< UnitId for logging */  
+        LoggingClass unitClass;       /**< UnitClass for logging */
     };
 
     typedef sequence<mEnvironmentComponent> mEnvironmentComponents;
@@ -163,6 +192,8 @@ module PlatformManagement
         code enableDevice();
         code disableDevice();
         code switchEnableDevice(out boolean on);
+        
+        code checkBindingConflict();
     };
 
     enum DataStatusExtension
@@ -195,32 +226,6 @@ module PlatformManagement
     };
 
 
-    /** Platform interface for the device components. Via this interface the device
-        components communicate with the platform manager internally. */
-    enum LoggingClass
-    {
-      LC_APPLICATION,
-      LC_PRINTER,
-      LC_SCANNER,
-      LC_CONVEYOR,
-      LC_BIN1,
-      LC_BIN2,
-      LC_BIN3,
-      LC_BIN4,
-      LC_CAPTURE_PRINTER,
-      LC_CAPTURE_READER,
-      LC_DISPENSER_PRINTER,
-      LC_DISPENSER_READER,
-      LC_PRINTER_FEEDER,
-      LC_CONVEYOR_FEEDER,
-      LC_DISK,
-      LC_NETWORK,
-      LC_UNDEFINED,
-      LC_PLF,
-      LC_INDICATOR,
-      LC_AUDIO
-    };
-
     struct LoggingComponentData
     {
       string componentRef;      // identify virtual component by virtualComponentRef
@@ -399,6 +404,7 @@ module PlatformManagement
             @param compList - The list of virtual components belonging to the device component
             @return RC_OK. */
         code bindDevice(in string realDeviceName, in ComponentIF comp, in mEnvironmentComponents compList);
+        code getBindingConflict(in string realDeviceName, out mEnvironmentComponents conflicted);
 
         /** Deletes a device component from the list of the current available device components.
             @param realDeviceName - The name of the device component
@@ -434,6 +440,9 @@ module PlatformManagement
         code enableDisplay();
         code disableDisplay();
         code switchEnableDisplay(out boolean on);
+        
+        code registerDevice(in string realDeviceName);
+        code deregisterDevice(in string realDeviceName);
     };
 };
 
-- 
2.41.0.windows.1

